acestep/training/data_module_test.py

```
@@ -0,0 +1,121 @@
+            PreprocessedTensorDataset("/tmp/nonexistent_xyz_12345")
+
+
+class LoadDatasetFromJsonTests(unittest.TestCase):
+    """Tests for load_dataset_from_json path validation."""
+
+    def setUp(self):
+        set_safe_root(tempfile.gettempdir())
+
+    def test_nonexistent_file_raises(self):
+        with self.assertRaises(ValueError):
+            load_dataset_from_json("/tmp/nonexistent_file.json")
```

Ruff (0.15.0)

[error] Probable insecure usage of temporary file or directory: "/tmp/nonexistent_file.json"

(S108)